# Исследование надежности заемщиков


## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
data['debt'].unique()

array([0, 1])

Данные в поле `debt` булевы - при 1 была задолжность по возврату кредитов, при 0 - отсутствовала. Для удобства визуального отображения сделаем столбец, который будет называется `is_debt` и иметь два значения - `exist` и `no_debt`.

In [23]:
data['is_debt'] = data['debt'].map({0: 'no_debt', 1: 'exist'})

In [24]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

Количество детей варьируется от 0 до 5. Для того чтобы увидеть зависимость, построим сводную таблицу количества детей от существования задолжности и посчитаем процент людей, имеющих задолжность по кредиту от общего количества людей с детьми для каждой категории.

In [25]:
def create_pivot_table(table, first_column, second_column='is_debt', value='debt'):
    pivot = table.pivot_table(index=first_column, columns=second_column, values=value, aggfunc='count', fill_value=0)
    pivot['percent'] = round((pivot['exist'] / (pivot['no_debt'] + pivot['exist'])) * 100, 2)
    return pivot

In [26]:
children_and_debt = create_pivot_table(data, 'children')
children_and_debt

is_debt   exist  no_debt  percent
children                         
0          1063    13028     7.54
1           444     4364     9.23
2           194     1858     9.45
3            27      303     8.18
4             4       37     9.76
5             0        9     0.00

На сводной таблице задолжностей для клиентов с пятью детьми не найдено. Проверим данные в таблице для пяти детей.

In [27]:
data.loc[data['children'] == 5, ['children', 'debt', 'is_debt']]

children  debt  is_debt
3979          5     0  no_debt
4397          5     0  no_debt
7866          5     0  no_debt
15822         5     0  no_debt
15916         5     0  no_debt
16211         5     0  no_debt
20452         5     0  no_debt
20837         5     0  no_debt
21156         5     0  no_debt

Действительно, семьи с пятью детьми не имеют задолжностей по кредитам. Поскольку количество клиентов с различным количеством детей разное, отсортируем полученную таблицу по проценту задолжности относительно группы клиентов с одним количеством детей.

In [28]:
children_and_debt.sort_values(by='percent', ascending=False)

is_debt   exist  no_debt  percent
children                         
4             4       37     9.76
2           194     1858     9.45
1           444     4364     9.23
3            27      303     8.18
0          1063    13028     7.54
5             0        9     0.00

Относительно группы больший процент не вернувших кредит среди клиентов с 4, 2 и 3 детьми, если делать оценку внутри группы. Однако обратив внимание на количество людей с 4, 3 и 5 детьми, можно сделать вывод, что количества не хватает для объективной оценки. Поэтому для поиска зависимости оставим категории людей с 2, 1 и 0 детей.

In [29]:
children_and_debt = children_and_debt.drop(labels=[4, 3, 5], axis=0)

In [30]:
children_and_debt.sort_values(by='percent', ascending=False)

is_debt   exist  no_debt  percent
children                         
2           194     1858     9.45
1           444     4364     9.23
0          1063    13028     7.54

**Вывод:** Наибольшее число людей, имеющих задолжность по кредиту среди клиентов, у которых нет детей. При оценке задолжности через отношение количества клиентов с задолжностью к общему количеству клиентов с группировкой по количеству детей больший процент людей, не вернувших кредит вовремя, растет с количеством детей для групп с 1 и 2 детьми соответственно.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Для начала рассмотрим данные для анализа: семейное положение - столбец `family_status` и возврат кредита в срок - столбец `debt`

In [31]:
data['family_status'].unique()

array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'Не женат / не замужем'], dtype=object)

In [32]:
data['debt'].unique()

array([0, 1])

Данные в поле `debt` булевы - при 1 была задолжность по возврату кредитов, при 0 - отсутствовала. Для удобства визуального отображения сделаем столбец, который будет называется `is_debt` и иметь два значения - `exist` и `no_debt`. 
Функция и данные были добавлены в предыдущем задании, поэтому заново проделывать подобные махинации нет смысла.

Создадим сводную таблицу для данных о семейном положении и статусе возврата кредита. Для оценки добавим колонку `percent`, которая поможет оценить количество задолжностей внутри различных семейных статусов.

In [33]:
family_status_and_debt = create_pivot_table(data, 'family_status')

In [34]:
family_status_and_debt.sort_values(by='exist', ascending=False)

is_debt                exist  no_debt  percent
family_status                                 
женат / замужем          927    11334     7.56
гражданский брак         385     3749     9.31
Не женат / не замужем    273     2523     9.76
в разводе                 84     1105     7.06
вдовец / вдова            63      888     6.62

In [35]:
family_status_and_debt.sort_values(by='percent', ascending=False)

is_debt                exist  no_debt  percent
family_status                                 
Не женат / не замужем    273     2523     9.76
гражданский брак         385     3749     9.31
женат / замужем          927    11334     7.56
в разводе                 84     1105     7.06
вдовец / вдова            63      888     6.62

**Вывод:** Наибольшее количество клиентов с задолжностью среди людей, которые находятся в браке. Однако при оценке процента людей задолжностью при группировке по статусу отношений, количество людей не вернувших долг во время, среди категорий `не женат/не замужем`, `гражданский брак` по убыванию соотвественно. Наименьший процент клиентов с задолжностью среди категорий `вдовец/вдова`, `в разводе`, `женат/замужем`, данные категории более надежны для выдачи кредита, категории указаны в порядке возрастания процента.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Для анализа необходимо оценить значения столбцов `total_income_category`, в котором указана категория и столбец `debt`, в котором при помощи булевых значений указано наличие задолжности. Для формирования сводной таблицы будем использовать созданный ранее столбец `is_dept`, где при помощи строковых значений указано наличие и отсутствие задолжности - данные `exist` и `no_debt` соответственно.

In [36]:
data['total_income_category'].unique()

array(['B', 'C', 'D', 'E', 'A'], dtype=object)

Для удобства анализа вспомним информацию о том, какие диапазоны сумм дохода соответсвуют каждой категории:
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.
Построем сводную таблицу категорий дохода от наличия задолжности

In [37]:
income_category_and_debt = create_pivot_table(data, 'total_income_category')
income_category_and_debt.sort_values(by='exist', ascending=False)

is_debt                exist  no_debt  percent
total_income_category                         
C                       1353    14568     8.50
B                        354     4660     7.06
D                         21      328     6.02
A                          2       23     8.00
E                          2       20     9.09

In [38]:
income_category_and_debt.sort_values(by='percent', ascending=False)

is_debt                exist  no_debt  percent
total_income_category                         
E                          2       20     9.09
C                       1353    14568     8.50
A                          2       23     8.00
B                        354     4660     7.06
D                         21      328     6.02

Для категорий `A`, `D`, `E` не достаточно данных для оценки возврата кредита в срок. Для простоты анализа уберем данные, не влияющие на итоговый вывод

In [39]:
income_category_and_debt = income_category_and_debt.drop(axis=0, labels=['E', 'A', 'D'])

In [40]:
income_category_and_debt.sort_values(by='percent', ascending=False)

is_debt                exist  no_debt  percent
total_income_category                         
C                       1353    14568     8.50
B                        354     4660     7.06

**Вывод:** Наибольшее количество задолжностей среди людей с категорией `С` - доход от 50001 до 200000 и по количественным, и по процентным метрикам. Данных для категорий `A`, `D`, `E` не достаточно для анализа. Более надежной группой для выдачи кредита будут клиенты с категорией дохода `B` - от 200001 до 1000000.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Для оценки возврата кредита в срок от цели кредита необходимо рассмотреть столбцы с данными `purpose_category` и столбец `debt`, в котором при помощи булевых значений указано наличие задолжности. Для формирования сводной таблицы будем использовать созданный ранее столбец `is_dept`, где при помощи строковых значений указано наличие и отсутствие задолжности - данные `exist` и `no_debt` соответственно.

In [41]:
data['purpose_category'].unique()

array(['операции с недвижимостью', 'операции с автомобилем',
       'получение образования', 'проведение свадьбы'], dtype=object)

Для получения отношения количества клиентов с задолжностью к общему количеству людей внутри категори добавим столбец `percent`

In [42]:
purpose_and_debt = create_pivot_table(data, 'purpose_category')

In [43]:
purpose_and_debt.sort_values(by='exist', ascending=False)

is_debt                   exist  no_debt  percent
purpose_category                                 
операции с недвижимостью    780     9971     7.26
операции с автомобилем      400     3879     9.35
получение образования       369     3619     9.25
проведение свадьбы          183     2130     7.91

In [44]:
purpose_and_debt.sort_values(by='percent', ascending=False)

is_debt                   exist  no_debt  percent
purpose_category                                 
операции с автомобилем      400     3879     9.35
получение образования       369     3619     9.25
проведение свадьбы          183     2130     7.91
операции с недвижимостью    780     9971     7.26

**Вывод:** наибольшее количество людей с задолжностью среди людей, которые берут кредит для операций с недвижимостью, однако при оценке возврата кредита в срок внутри категорий, больший процент людей не возвращают кредит во время среди категорий `операций с автомобилем`, `получение образования`, если оценивать процент задолжностей как отношение должников к общему количеству людей внутри категории.

На основани этого можно сделать вывод, что клиенты с целью взятия кредита на `проведение свадьбы` и `операции с недвижимостью` более надежны. Тогда как риск возникновения долга среди клиентов с целями `операции с автомобилем`, `получение образования` повышается.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Причины появления пропуска данных могут быть различные - поле являлось не обязательным для заполнения в анкете и человек не стал указывать. Так же пропуски могут быть следствием ошибок при выгрузке данных в scv файл или на аналогичном этапе.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* поскольку медианное значение дает более объективную оценку среднего, исключая влияние выбросов значений, которые могут находится на краях диапазона и тем самым влиять на среднее значение.

### Шаг 4: общий вывод.

В ходе анализа была проведена работа с данными, включающая в себя:

- Заполнение пропусков данных медианным значением в столбцах `days_employed`, `total_income`, отвечающие за общий трудовой стаж в днях и ежемесячный доход соответственно
- Поиск и удаление аномалий в данных - удаление строк, в которых в столбце `children` указано отрицательное количество детей и аномально большое: 20. Для аномалий в виде отрицательного стажа работы в столбце `days_employed` была произведена замена на положительные значения
- Для `education` — уровень образования клиента было произведено приведение к единообразному представлению данных, которые являются неявными дубликатами и могут повлиять на дальшнейший анализ
- Чтобы произвести оценку зависимостей дохода от задолжностей было проведено категоризирование данных на пять категорий в зависимости от уровня дохода
- Для дальнейшего анализа и единообразности данных было произведено категоризирование данных `purpose` — цель получения кредита

Ниже для иллюстрации зависимостей представлены таблицы категорий от наличия задолжностей. 

Описание данных в таблицах: 
- `exist` - имеется задолжность, количество клиентов
- `no_debt` - отсутствие задолжности, количество клиентов
- `percent` - отношение количества клиентов с задолжностью к общему количеству людей в категории в процентах

<b>Зависимость между количеством детей и возвратом кредита в срок </b>

Наибольшее количество должников выявленно среди категории клиентов без детей. Однако доля должников увеличивается с увеличением количества детей (1-2)

In [45]:
children_and_debt.sort_values(by='percent', ascending=False)

is_debt   exist  no_debt  percent
children                         
2           194     1858     9.45
1           444     4364     9.23
0          1063    13028     7.54

<b>Зависимость между семейным положением и возвратом кредита в срок </b>

Наибольшая доля должников среди людей, не находящихся в зарегистрированном браке: `Не женат/не замужем`- 9.76%, `Гражданский брак`- 9.31%. Более надежными являются категории людей, уже побывавших в браке: `в разводе` - 7.06%
`вдовец/вдова` - 6.62%. Для данных групп возможно влияние возраста


In [46]:
family_status_and_debt.sort_values(by='percent', ascending=False)

is_debt                exist  no_debt  percent
family_status                                 
Не женат / не замужем    273     2523     9.76
гражданский брак         385     3749     9.31
женат / замужем          927    11334     7.56
в разводе                 84     1105     7.06
вдовец / вдова            63      888     6.62

<b>Зависимость между уровнем дохода и возвратом кредита в срок </b>

Наибольшее количество и наибольшая доля задолжностей среди людей с категорией С - доход от 50001 до 200000. Данных для категорий A, D, E не достаточно для анализа. Более надежной группой для выдачи кредита будут клиенты с категорией дохода B - от 200001 до 1000000.


In [47]:
income_category_and_debt.sort_values(by='percent', ascending=False)

is_debt                exist  no_debt  percent
total_income_category                         
C                       1353    14568     8.50
B                        354     4660     7.06

<b>Зависимость между целями кредита и возвратом кредита в срок </b>

После проведения анализа наиболее рискованными категориями оказались `операции с автомобилем`, `получение образования` - доля должников для обеих категорий в среднем 9.3%. Более надежными для выдачи кредита являются категории с целями `проведение свадьбы` и `операции с недвижимостью`.

In [48]:
purpose_and_debt.sort_values(by='percent', ascending=False)

is_debt                   exist  no_debt  percent
purpose_category                                 
операции с автомобилем      400     3879     9.35
получение образования       369     3619     9.25
проведение свадьбы          183     2130     7.91
операции с недвижимостью    780     9971     7.26

<b>Рекомендации</b>

Наиболее надежная категория людей для выдачи кредита - клиент, состоящий в браке или побывавший в нем, не имеющий детей, с доходом в диапазоне 200001 до 1000000 и целью кредита `проведение свадьбы` или `операции с недвижимостью`.